python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_resources.py

```diff
@@ -73,3 +75,30 @@ def test_refresh_access_token(base_api_mocks: responses.RequestsMock) -> None:
         assert access_token_call_body["client_id"] == TEST_CLIENT_ID
         assert access_token_call_body["client_secret"] == TEST_CLIENT_SECRET
         assert jobs_api_call.request.headers["Authorization"] == f"Bearer {TEST_ACCESS_TOKEN}"
+
+
+def test_basic_resource_request(
+    fetch_workspace_data_api_mocks: responses.RequestsMock,
+) -> None:
+    resource = AirbyteCloudWorkspace(
+        workspace_id=TEST_WORKSPACE_ID,
+        client_id=TEST_CLIENT_ID,
+        client_secret=TEST_CLIENT_SECRET,
+    )
+    client = resource.get_client()
+
+    # fetch workspace data calls
+    client.get_connections()
+    client.get_connection_details(connection_id=TEST_CONNECTION_ID)
+    client.get_destination_details(destination_id=TEST_DESTINATION_ID)
+
+    assert len(fetch_workspace_data_api_mocks.calls) == 4
+    # The first call is to create the access token
+    assert "Authorization" not in fetch_workspace_data_api_mocks.calls[0].request.headers
+    # The two next calls are actual API calls
+    assert "connections" in fetch_workspace_data_api_mocks.calls[1].request.url
+    assert "connections/get" in fetch_workspace_data_api_mocks.calls[2].request.url
+    assert TEST_CONNECTION_ID in fetch_workspace_data_api_mocks.calls[2].request.body.decode()
```

Comment:
having trouble interpreting these assertions. Can we do a few utility things:
add a helper method to tease out the url: `mock_url(mock_call)`
assert against a constant that describe what the actual url should be.
seems like the first api call is always going to be auth, and this seems like a really backward way of testing it. We're just testing that Auth wasn't set in the first call, not that we're setting auth. Can we have a shared utility for that as well, and assert it more directly? Something like:
```
def assert_and_split_calls(calls):
    assert calls[0].url == ...
    assert auth in client or whatever
    return calls[1:]
```

Reply:
That's a good call - done in [9f8c99d](https://github.com/dagster-io/dagster/pull/26242/commits/9f8c99d4cd48bc0bc85b33a8e5f3a5e8e7042e9a)